starlite/cache/redis_cache_backend.py

```diff
@@ -0,0 +1,99 @@
+from typing import Any, Optional
+
+from pydantic import BaseModel
+
+from starlite.exceptions import MissingDependencyException
+
+try:
+    from redis.asyncio import Redis
+    from redis.asyncio.connection import ConnectionPool
+except ImportError as e:
+    raise MissingDependencyException(
+        "To use starlite.redis_cache_backend, install starlite with 'redis_cache_backend' extra, e.g. `pip install starlite[redis_cache_backend]`"
+    ) from e
+
+
+from starlite.cache.base import CacheBackendProtocol
+
+
+class RedisCacheBackendConfig(BaseModel):
+    """Redis cache backend configuration."""
+
+    url: str
```

Comment:
the type here can be `RedisDsn` from pydantic

Reply:
@Goldziher what is the proper way to initialize `RedisDsn`?

- When I do this:
   ```python
   url = RedisDsn("redis://localhost/")
   ```
   I get an error when running the tests:
   ```text
   E   pydantic.error_wrappers.ValidationError: 1 validation error for RedisCacheBackendConfig
   E   URL
   E     invalid or missing URL scheme (type=value_error.url.scheme)
   ```

- And when I do this:
   ```python
   url: RedisDsn = "redis://localhost/"
   ```
   I get a type-checker error:
   ```text
   error: Incompatible types in assignment (expression has type "str", variable has type "RedisDsn")
   ```

In the end of the day it should be a simple string...

Reply:
Yup.

Reply:
Can you help with this? https://github.com/starlite-api/starlite/pull/564#issuecomment-1274105021

I'm not sure why the CI fails...

Reply:
I can fix and merge l8r, will have to see whats going on